Use an EIP for the OpenVPN server's public IP address

jsf9k opened the pull request:
## 🗣 Description

In this pull request I change the Terraform code to use an EIP for the public IP of the OpenVPN server.

## 💭 Motivation and Context

This change allows the OpenVPN server to keep the same public IP address when the EC2 instance is stopped and started.  Previously the public IP address would change when this happened.

## 🧪 Testing

I have used these changes to deploy OpenVPN to our staging COOL environment, and I have tested that the public address indeed remains the same after stopping and starting the instance.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Modify code to use an EIP for the OpenVPN server

jsf9k commented:
Given that [the Teraform code for the A record](https://github.com/cisagov/openvpn-server-tf-module/blob/develop/route53.tf#L1-L8) does not take into account the `associate_public_ip_address` input variable, I think it makes sense to remove it altogether and just always define a public IP address for the OpenVPN server.  What do you think @dav3r and @felddy?

hillaryj reviewed (approved):
Looking good! I think it makes a lot of sense to define the public ip then.

🥇

dav3r commented:
> Given that [the Teraform code for the A record](https://github.com/cisagov/openvpn-server-tf-module/blob/develop/route53.tf#L1-L8) does not take into account the `associate_public_ip_address` input variable, I think it makes sense to remove it altogether and just always define a public IP address for the OpenVPN server. What do you think @dav3r and @felddy?

Agreed! 👍

dav3r reviewed (approved):
Excellent! 🐘

jsf9k committed:
Remove associate_public_ip_address variable

This variable isn't referenced when creating the A record, so we may
as well just _always_ give the OpenVPN server a public IP address.

jsf9k committed:
Update README with latest terraform-docs content

We needed to re-run terraform-docs anyway after removing a variable in
the last commit.

mcdonnnj commented on README.md:
```diff
@@ -48,22 +48,23 @@ module "example" {
-| template | n/a |
+| aws | ~> 2.0 |
+| aws.dns | ~> 2.0 |
+| template | ~> 2.0 |
```
I know it is `template = "~> 2.0"` in this repo, but in [cisagov/cool-assessment-terraform](https://github.com/cisagov/cool-assessment-terraform/blob/cf132c570f5d0cbf648c3ff2c78355653fecc87f/versions.tf#L10) and [cisagov/cool-sharedservices-nessus](https://github.com/cisagov/cool-sharedservices-nessus/blob/e40dc853509a395e96b43dba7cc360a99b957cfd/versions.tf#L10) it is `2.1`. Should we standardize versions?

jsf9k replied:
Technically we should possibly be doing something like
```hcl
terraform {
  # We want to hold off on 0.13 until we have tested it.
  required_version = "~> 0.12.29"

  # If you use any other providers you should also pin them to the
  # major version currently being used.  This practice will help us
  # avoid unwelcome surprises.
  required_providers {
    aws      = "~> 2.70"
    template = "~> 2.1"
  }
}
```
since I think the `~> X.Y` syntax sets a minimum version of `X.Y` and a maximum version of "whatever is the latest release with major version `X`".  Practically I don't think it matters, since we always run `terraform init -upgrade`.  Hence `~>X.0` and `~>X.1` are indistinguishable for our purposes.

Note that [this page](https://www.terraform.io/docs/configuration/provider-requirements.html#best-practices-for-provider-versions) indicates that we should not be specifying maximum versions at all for modules that are not intended to be used as root modules.

I'm unclear what the best way forward is, but I'm fine with just bumbling ahead until we figure out what works best for us.  In the interim, I will change the template provider's version constraint to `~> 2.1`.

jsf9k replied:
See commit e0de5b5.

mcdonnnj replied:
Ah, right, root modules. I'm fine with however we want to forge ahead with at this time. I just remembered seeing a `template ~> 2.1` in another batch of PRs and thought I'd mention the different versions.

jsf9k replied:
I'm going to punt on this comment for now, but I do think we should revisit it after doing some more noodling.  CRITICAL PATH!

mcdonnnj reviewed (approved):
Makes sense. Please see my one question though.

dav3r reviewed (approved):
👍 👍

jsf9k committed:
Tweak template provider's version constraint

This is done to be consistent with other repositories.
